tests/models/test_boundary_condition.py

```diff
@@ -113,57 +113,97 @@ def test_boundary_condition_mixin(t_end: int):
+        and Robin approximating Dirichlet on the north/south boundaries. This method
+        returns the indices of the north and south boundaries.
+
+        """
+        bounds = self.domain_boundary_sides(g)
+        return bounds.north + bounds.south
+
+    def dir_inds(self, g):
+        """Indices for the Dirichlet boundaries for test.
+
+        The Robin limit case test tests setups with Dirichlet on west/east boundaries,
+        and Robin approximating Dirichlet on the north/south boundaries. This method
+        returns the indices of the east and west boundaries.
+
+        """
+        bounds = self.domain_boundary_sides(g)
+        return bounds.west + bounds.east
 
     def bc_values_displacement(self, bg: pp.BoundaryGrid) -> np.ndarray:
-        """Assigns displacement values in the x-direction of the west boundary."""
-        values = np.ones((self.nd, bg.num_cells))
-        bounds = self.domain_boundary_sides(bg)
-        values[0][bounds.west] += np.ones(len(values[0][bounds.west]))
+        """Assigns displacement values in the x-direction of the west and east
+        boundaries."""
+        values = np.zeros((self.nd, bg.num_cells))
+        values[0, self.dir_inds(bg)] = 42
         return values.ravel("F")
 
-    def bc_values_pressure(self, bg: pp.BoundaryGrid) -> np.ndarray:
-        """Assigns pressure values on the west boundary."""
-        values = np.ones(bg.num_cells)
-        bounds = self.domain_boundary_sides(bg)
-        values[bounds.west] += np.ones(len(values[bounds.west]))
+    def _bc_values_scalar(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Assigns scalar values on the west and east boundaries."""
+        values = np.zeros(bg.num_cells)
+        values[self.dir_inds(bg)] = 42
         return values
 
+    def bc_values_pressure(self, bg: pp.BoundaryGrid) -> np.ndarray:
+        """Assigns pressure boundary values."""
+        return self._bc_values_scalar(bg=bg)
+
     def bc_values_temperature(self, bg: pp.BoundaryGrid) -> np.ndarray:
-        """Assigns temperature values on the west boundary."""
-        values = np.ones(bg.num_cells)
-        bounds = self.domain_boundary_sides(bg)
-        values[bounds.west] += np.ones(len(values[bounds.west]))
-        return values
+        """Assigns temperature boundary values."""
+        return self._bc_values_scalar(bg=bg)
 
 
-class BCDirNeu:
-    """Set Dirichlet and Neumann for momentum balance and mass and energy balance.
+class BCRobin:
+    """Set Dirichlet and Robin for momentum balance and mass and energy balance.
+
+    Sets Dirichlet on the west and east boundaries, and Robin on the top and bottom
+    boundaries. The value of the Robin weight is determined from the parameter "alpha"
+    in the params dictionary. This class also sets Robin boundary values.
 
-    Sets Dirichlet on the west boundary, and Neumann on all other boundaries.
+    This class is common for all the model classes that enters into testing Robin limit
```

Comment:
Is that true? Isn't it only used for the Robin version, not the reference model?

Reply:
You're correct, will fix!